platform/mellanox/mlnx-platform-api/sonic_platform/modules_mgmt.py

```diff
@@ -0,0 +1,784 @@
+                logger.log_info("timerTask working on port {}".format(port))
+                empty = False
+                module = self.sfp_port_dict[port]
+                logger.log_info("timerTask got module with port_num {} from port {}".format(module.port_num, port))
+                state = module.get_current_state()
+                if module and state == STATE_NOT_POWERED:
+                    logger.log_info("timerTask module {} current_state {} counting seconds since reset_start_time"
+                                  .format(module, module.get_current_state()))
+                    if time.time() - module.reset_start_time >= 3:
+                        # set next state as STATE_POWERED state to trigger the function of check module type
+                        logger.log_info("timerTask module port {} locking lock of port {}".format(module.port_num, module.port_num))
+                        self.modules_lock_list[module.port_num].acquire()
+                        logger.log_info("timerTask module port {} setting next state to STATE_POWERED".format(module.port_num))
+                        module.set_next_state(STATE_POWERED)
+                        logger.log_info("timerTask module port {} advancing next state".format(module.port_num))
+                        module.advance_state()
+                        logger.log_info("timerTask module port {} releasing lock of port {}".format(port, module.port_num))
+                        self.modules_lock_list[module.port_num].release()
+                        logger.log_info("timerTask module port {} adding to delete list to remove from waiting_modules_list".format(module.port_num))
+                        port_list_to_delete.append(module.port_num)
+            logger.log_info("timerTask deleting ports {} from waiting_modules_list...".format(port_list_to_delete))
+            for port in port_list_to_delete:
+                logger.log_info("timerTask deleting port {} from waiting_modules_list".format(port))
+                self.waiting_modules_list.remove(port)
+            logger.log_info("timerTask waiting_modules_list after deletion: {}".format(self.waiting_modules_list))
+            time.sleep(1)
+            i += 1
+
+    def get_sysfs_ethernet_port_fd(self, sysfs_fd, port):
+        sysfs_eth_port_fd = sysfs_fd.format(port)
+        return sysfs_eth_port_fd
+
+    def add_port_to_wait_reset(self, module_sm_obj):
+        module_sm_obj.reset_start_time = time.time()
+        logger.log_info("add_port_to_wait_reset reset_start_time {}".format(module_sm_obj.reset_start_time))
+        module_sm_obj.wait_for_power_on = True
+        logger.log_info("add_port_to_wait_reset wait_for_power_on {}".format(module_sm_obj.wait_for_power_on))
+        self.waiting_modules_list.add(module_sm_obj.port_num)
+        logger.log_info("add_port_to_wait_reset waiting_list after adding: {}".format(self.waiting_modules_list))
+
+    def add_ports_state_to_state_db(self, dynamic=False):
```

Comment:
@dbarashinvd we discussed this to make use of TRANSCIEVER_INFO table to know CMIS vs non-CMIS.

Reply:
done. please review again and approve